run.py

```diff
@@ -621,6 +625,7 @@ def check(self):
                 for check_count, check in enumerate(self.config["checks"]):
                     for key in check:
                         if key in ["filter", "shell", "stats"]:
+                            self.reset_count(count)
```

Comment:
What has this to do with the goal of this PR, ie Give test count as output rather than check count

Reply:
Let's say there's a test `SampleTest` with 4 checks `check1`, `check2`, `check3` and `check4`.
Previously (before the checks count was introduced), the structure of the code was such that if `check2` failed, the code would return and the entire test will show up as failed without running the `check3` and `check4`.
However, while check count was introduced, the actual request was to be able to run all the checks and point out exactly the check which failed so in that that if `check2` failed, `check3` and `check4` would still run and the output would clearly point that `check2` failed and not all the other checks. In order to implement this, every check's state was to be recorded i.e. whether it `failed`, `passed` or was `skipped`. Alongwith this, I was storing the count of states of the checks to be able to give a final result (which was check count). Now that state is reset (the line above) with every test as we're only considering the count of tests in the final output.
The state could be made boolean but that would also have to be reset on ever test.

Reply:
I am not sure I understand all of this code (maybe I need more coffee)
So, it may deserve some more comments.

Why is `self.reset_count(count)` called on every iteration of the loop `for key in check:` ?
So that count has only one key whose value is 1 (and not more) ?

Maybe my confusion comes from the fact that : ok we should have he final stats about tests... but I would also expect to have detailed stats about checks, especially when running only one test...

Reply:
@catenacyber that was a bug. Thanks for finding this out. You work great even without coffee. :)